crawl-ref/source/skills.cc

```diff
@@ -1169,34 +1170,35 @@ pair<int,int> skill_level_to_diffs(skill_type skill, double amount, int scaled_t
         if (reduced && you_skill_cost_level)
             ++max_xp;
 
-        int cost = calc_skill_cost(you_skill_cost_level);
+        const int cost = calc_skill_cost(you_skill_cost_level);
         // Maximum number of skill points to transfer in one go.
         // It's max_xp*10/cost rounded up.
-        int max_skp = (max_xp * 10 + cost - 1) / cost;
-        max_skp = max(max_skp, 1);
-        int delta_skp = min<int>(abs((int)(target - you_skill)),
+        const int max_skp = max((max_xp * 10 + cost - 1) / cost, 1);
+        skill_diff delta;
```

Comment:
I consider that but I don't think that works for this use because the xp value is actually dependent on the skill point value, though maybe c++ constructors could handle that.